### PR TITLE
Add Vuex to our project

### LINKED ISSUE #29: Separate API/Firebase calls into a separate file
After working on some Android architecture, I have begun to realize the importance of software architecture. We should separate out all the API/Firebase calls in our files to a separate and centralized file.

Proposed:

- make a `network` folder
- make an `API.ts` file
- consider making [data transfer object](https://en.wikipedia.org/wiki/Data_transfer_object) files, which will make changing models (#12) on the front end much easier in the future

### PR DESCRIPTION
**Pull request by kungpaogao**
### Summary 

This pull request is the *first step* towards adding [Vuex](https://vuex.vuejs.org/) to our project. This is significant for reasons described in #35. Basically, it allows us to separate the state of our app from the components, which is important especially because we have many form components modifying the same data.

**Completed**

- set up framework for creating Vuex modules
- link Inventory to use store
- added Vuexfire to bind state to Firestore
- established framework for unidirectional flow with `src/network` folder
  - fixes #29 
- updated npm packages, ran `npm audit`
- added documentation to files that were modified in this PR

### Test Plan 

Application builds and behaves as expected. Separate components are able to react to the same state, and unidirectional flow works as expected (see `Admin.vue`).

If this passes CI, we should be good for merging.

### Breaking Changes  

This fundamentally changes how we should think about state in our application. While simple components should still interact with each other using [props](https://vuejs.org/v2/guide/components.html#Passing-Data-to-Child-Components-with-Props) and [events](https://vuejs.org/v2/guide/components.html#Listening-to-Child-Components-Events), most of the inventory and form management will require using Vuex.

I recommend reading through the resources listed down below to get a good mental model of what Vuex does for our application. The official documentation and the video in it is particularly good for getting a basic idea.

### Next Steps 

Our use of Vuex is not complete. As of this PR, only the Inventory and Admin views are linked to the Vuex store.

Tasks:

- update all `EditCard` components to use the store
- add documentation for Vuex functions
- add style guide for writing Vuex modules for future features

### Other Notes 

Resources used throughout making progress for this PR:

- official docs: https://vuex.vuejs.org/
- quick start guides: 
  - https://medium.com/dailyjs/mastering-vuex-zero-to-hero-e0ca1f421d45
  - https://scotch.io/tutorials/state-management-in-vue-getting-started-with-vue#toc-the-challenge-of-state-management
- using Vuex with class components (what we use):
  - https://github.com/vuejs/vue-class-component/issues/56
  - https://github.com/vuejs/vue-class-component/issues/109
- Vuex style guide (from GitLab): https://docs.gitlab.com/ee/development/fe_guide/vuex.html
- Vuex with TypeScript: https://codeburst.io/vuex-and-typescript-3427ba78cfa8?gi=f52ed2c38ac4
- Vuexfire (used for Firestore bindings): https://vuefire.vuejs.org/vuexfire/
- an example: https://github.com/vuejs/vuex/tree/dev/examples/shopping-cart

### CODE REVIEW

File: src/views/Inventory.vue [hunk 1]
```diff
@@ -42,58 +47,80 @@
...
+  methods: mapActions(namespace, [
+    "bindInventory",
+    "setCurrent",
+    "clearCurrent",
+  ]),
 })
 export default class Inventory extends Vue {
-  status = Status;
+  current!: Furniture;
+
```
Comment (rayrayweng) [inline src/views/Inventory.vue:77]: Would you need to also define inventory down here as well?

Comment (kungpaogao) [reply src/views/Inventory.vue:77]: Ah, good question! If I wanted to use the `inventory` variable within the class code below, I would have to declare it as a property within the class. But, I only use it in the `<template>` above, so I don't need it.

Basically, repeating the declaration inside the class is only needed in order for functions/stuff inside the class to use it.

Comment (rayrayweng) [reply src/views/Inventory.vue:77]: Gotcha. And for my own understanding, you used inventory in the v-data-table part?

Comment (kungpaogao) [reply src/views/Inventory.vue:77]: yup

**Pull request merged**